pkg/util/cel/cel_compilation_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"testing"
+)
+
+func TestCelCompilation(t *testing.T) {
+	cases := []struct {
+		name      string
+		input     spec.Schema
+		expr      string
+		wantError bool
+	}{
+		{
+			name: "valid object",
+			input: spec.Schema{
+				SchemaProps: spec.SchemaProps{
+					Type: []string{"object"},
+					Properties: map[string]spec.Schema{
+						"minReplicas": {
+							SchemaProps: spec.SchemaProps{
+								Type: []string{"integer"},
+							},
+						},
+						"maxReplicas": {
+							SchemaProps: spec.SchemaProps{
+								Type: []string{"integer"},
+							},
+						},
+					},
+				},
+			},
+			expr:      "minReplicas < maxReplicas",
+			wantError: false,
+		},
+		{
+			name: "invalid type",
+			input: spec.Schema{
+				SchemaProps: spec.SchemaProps{
+					Type: []string{"nonSupported"},
+				},
+			},
+			expr:      "unsupported type check",
+			wantError: true,
```

Comment:
Need to verify that the error is the one you expect here.

Also need to make sure to cover the various complication errors that can happen.